cadcutils/cadcutils/net/ws.py

```diff
@@ -549,32 +550,37 @@ def combine_headers(new_headers):
                         url,
                         data=reader,
                         verify=self.verify, **kwargs)
-                trans_id = response.headers[PUT_TXN_ID]
+                trans_id = response.headers.get(PUT_TXN_ID, None)
                 # check the file made it OK
                 dest_md5 = net.extract_md5(response.headers)
                 if dest_md5 != reader.md5_checksum:
                     msg = 'File {} not properly uploaded. ' \
                           'Mismatched md5 src vs dest: {} vs {}'.format(
                             src, reader.md5_checksum, dest_md5)
                     self.logger.warning(msg)
-                    # abort transaction
-                    kwargs[HEADERS] = combine_headers(
-                        {PUT_TXN_ID: trans_id,
-                         PUT_TXN_OP: PUT_TXN_ABORT,
-                         HTTP_LENGTH: '0'})
-                    self._get_session().post(url, verify=self.verify, **kwargs)
+                    if trans_id:
```

Comment:
Does the service ever send back a transaction id of '0'?

Reply:
Good point. Extremely unlikely but I've updated it for correctness sake.